src/videoframeprocessor/DefaultVideoTransformDevice.ts

```diff
@@ -0,0 +1,224 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import DefaultBrowserBehavior from '../browserbehavior/DefaultBrowserBehavior';
+import Device from '../devicecontroller/Device';
+import VideoTransformDevice from '../devicecontroller/VideoTransformDevice';
+import Logger from '../logger/Logger';
+import DefaultVideoFrameProcessorPipeline from './DefaultVideoFrameProcessorPipeline';
+import DefaultVideoTransformDeviceObserver from './DefaultVideoTransformDeviceObserver';
+import VideoFrameProcessor from './VideoFrameProcessor';
+import VideoFrameProcessorPipeline from './VideoFrameProcessorPipeline';
+import VideoFrameProcessorPipelineObserver from './VideoFrameProcessorPipelineObserver';
+
+/**
+ * [[DefaultVideoTransformDevice]] is an augmented [[VideoInputDevice]].
+ * It transform the input {@link Device} with an array of {@link VideoFrameProcessor} to produce a `MediaStream`.
+ */
+export default class DefaultVideoTransformDevice
+  implements VideoTransformDevice, VideoFrameProcessorPipelineObserver {
+  private inputMediaStream: MediaStream;
+  private observers: Set<DefaultVideoTransformDeviceObserver> = new Set<
+    DefaultVideoTransformDeviceObserver
+  >();
+
+  constructor(
+    private logger: Logger,
+    private device: Device,
+    private processors: VideoFrameProcessor[],
+    private browserBehavior: DefaultBrowserBehavior = new DefaultBrowserBehavior(),
+    private pipe: DefaultVideoFrameProcessorPipeline | null = null
```

Comment:
Why do you need to take a pipeline as an argument, when you're taking a list of processors as an argument? Just create the pipeline once, and pass the processors to its constructor.

Reply:
now pipeline constructor takes processors as arguments.